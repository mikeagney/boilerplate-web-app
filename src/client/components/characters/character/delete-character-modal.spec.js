import React from 'react';
import { Modal } from 'react-bootstrap';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DeleteCharacterModal from './delete-character-modal';

describe('Modal dialog box for deleting a character', () => {
  const defaultProps = {
    name: 'John Doe',
    show: true,
    onCancel: jest.fn(),
    onConfirm: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('will render normally if show is false', () => {
    // Arrange
    const props = defaultProps;

    // Act
    const wrapper = shallow(<DeleteCharacterModal {...props} />);

    // Assert
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('will render hidden if show is false', () => {
    // Arrange
    const props = {
      ...defaultProps,
      show: false,
    };

    // Act
    const wrapper = shallow(<DeleteCharacterModal {...props} />);

    // Assert
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('will call onCancel when closed', () => {
    // Arrange
    const props = defaultProps;
    const wrapper = shallow(<DeleteCharacterModal {...props} />);

    // Act
    const onHide = wrapper.find(Modal).prop('onHide');
    onHide();

    // Assert
    expect(props.onCancel).toHaveBeenCalledTimes(1);
    expect(props.onConfirm).not.toHaveBeenCalled();
  });

  it('will call onCancel when cancel button is clicked', () => {
    // Arrange
    const props = defaultProps;
    const wrapper = shallow(<DeleteCharacterModal {...props} />);

    // Act
    wrapper.find('.delete-button-cancel').simulate('click');

    // Assert
    expect(props.onCancel).toHaveBeenCalledTimes(1);
    expect(props.onConfirm).not.toHaveBeenCalled();
  });

  it('will call onConfirm when confirm button is clicked', () => {
    // Arrange
    const props = defaultProps;
    const wrapper = shallow(<DeleteCharacterModal {...props} />);

    // Act
    wrapper.find('.delete-button-confirm').simulate('click');

    // Assert
    expect(props.onCancel).not.toHaveBeenCalled();
    expect(props.onConfirm).toHaveBeenCalledTimes(1);
  });
});
