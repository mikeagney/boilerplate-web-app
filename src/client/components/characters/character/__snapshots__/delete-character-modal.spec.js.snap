// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal dialog box for deleting a character will render hidden if show is false 1`] = `
<Bootstrap(Modal)
  onHide={[MockFunction]}
  show={false}
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      Confirm Delete
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    Really delete character John Doe?
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      className="delete-button-cancel"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
      variant="secondary"
    >
      Keep
    </Button>
    <Button
      active={false}
      className="delete-button-confirm"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
      variant="primary"
    >
      Delete
    </Button>
  </ModalFooter>
</Bootstrap(Modal)>
`;

exports[`Modal dialog box for deleting a character will render normally if show is false 1`] = `
<Bootstrap(Modal)
  onHide={[MockFunction]}
  show={true}
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      Confirm Delete
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    Really delete character John Doe?
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      className="delete-button-cancel"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
      variant="secondary"
    >
      Keep
    </Button>
    <Button
      active={false}
      className="delete-button-confirm"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
      variant="primary"
    >
      Delete
    </Button>
  </ModalFooter>
</Bootstrap(Modal)>
`;
